docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
 			description: "Raised when the provided input cannot be parsed."
 		}
 	}
+
+	// TRL type system
+	types: [TypeName=string]: {
+		#Use: "parameter" | "return"
+
+		description: string
+		use: [#Use, ...#Use]
+	}
+
+	types: {
+		"any": {
+			description: "A stand-in for any type."
+			use: ["parameter"]
+		}
+		"array": {
+			description: "A list of items."
+			use: ["parameter", "return"]
+		}
+		"boolean": {
+			description: "`true` or `false`."
+			use: ["parameter", "return"]
+		}
+		"float": {
+			description: "A 64-bit floating-point number."
+			use: ["parameter", "return"]
+		}
+		"map": {
+			description: """
+				A key-value map in which keys are strings and values can be of any TRL type,
+				including maps.
+				"""
+			use: ["parameter", "return"]
+		}
+		"integer": {
+			description: "A 64-bit integer."
+			use: ["parameter", "return"]
+		}
+		"null": {
+			description: "No value."
+			use: ["return"]
+		}
+		"path": {
+			description: "An event field."
+			use: ["parameter"]
+		}
+		"regex": {
+			description: "A regular expression."
+			use: ["parameter"]
+		}
+		"string": {
+			description: """
+				A sequence of characters. Remap converts strings in scripts to [UTF-8](\(urls.utf8))
+				and replaces any invalid sequences with `U+FFFD REPLACEMENT CHARACTER` (�).
+				"""
+			use: ["parameter", "return"]
+		}
```

Comment:
We should cover escaping here. I'm also curious where you might put string literals if/when we add them?

Reply:
Good question. I think that given that they aren't a separate type but rather one *way* of providing string values I think we should explain that here within the string type.

Reply:
Yeah, I think it might be better covered in the syntax section since it's really just a syntactical difference.